pkg/middleware/auth_proxy.go

```diff
@@ -31,24 +38,38 @@ func initContextWithAuthProxy(store *remotecache.RemoteCache, ctx *m.ReqContext,
 
 	// Check if allowed to continue with this IP
 	if result, err := auth.IsAllowedIP(); !result {
-		ctx.Logger.Error("auth proxy: failed to check whitelisted ip addresses", "message", err.Error(), "error", err.DetailsError)
+		logger.Error(
+			"Failed to check whitelisted IP addresses",
+			"message", err.Error(),
+			"error", err.DetailsError,
+		)
 		ctx.Handle(407, err.Error(), err.DetailsError)
 		return true
 	}
 
 	// Try to log in user from various providers
 	id, err := auth.Login()
 	if err != nil {
-		ctx.Logger.Error("auth proxy: failed to login", "message", err.Error(), "error", err.DetailsError)
-		ctx.Handle(500, err.Error(), err.DetailsError)
+		logger.Error(
+			"Failed to login",
+			"username", username,
+			"message", err.Error(),
+			"error", err.DetailsError,
+		)
+		ctx.Handle(407, err.Error(), err.DetailsError)
 		return true
 	}
 
 	// Get full user info
 	user, err := auth.GetSignedUser(id)
 	if err != nil {
-		ctx.Logger.Error("auth proxy: failed to get signed in user", "message", err.Error(), "error", err.DetailsError)
-		ctx.Handle(500, err.Error(), err.DetailsError)
+		logger.Error(
+			"Failed to get signed user",
+			"username", username,
+			"message", err.Error(),
+			"error", err.DetailsError,
+		)
+		ctx.Handle(407, err.Error(), err.DetailsError)
```

Comment:
Shouldn't this still be a `500`? We were just successful in logging in the user in the previous call (`auth.Login()`), seems to me that failure here is very unexpected.

Reply:
If user is removed from grafana db it will not going to be removed from cache until elapsed time for this entry is reached. Foremost `Login` will get the `id` from the cache then just pass it to this branch of code which will never find it.

I believe this is how issue #18439 came to be